File: Civitas - Ruby/lib/test_p4.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

require './vista_textual'
require './Controlador'
require './jugador_especulador'
require './jugador'
require './titulo_propiedad'

module Civitas
  puts "TEST P4 __________________________________"
  
  jugador = Jugador.new("Paco")
  jugador.propiedades << Titulo_propiedad.new("escarbuto", 0, 0, 0, 0, 0)

  puts "ANTES"
  puts jugador.to_s

  puts "DESPUES"
  jugador = Jugador_especulador.nuevo_especulador(jugador, 50)
  puts jugador.to_s
  puts jugador.propiedades[0].to_s

end

File: Civitas - Java/src/civitas/TestP4.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;

import java.util.ArrayList;
import GUI.CSO;

/**
 *
 * @author juanc
 */
public class TestP4 {

    public static void main(String[] args) {
        System.out.println("TEST P4 __________________________________");
        
        ArrayList<Jugador> todos = new ArrayList<>();
        
        todos.add(new Jugador("Paco")); 
        
        todos.get(0).getPropiedades().add( new TituloPropiedad("escarbuto", 0, 0, 0, 0, 0)  );
        System.out.println( todos.get(0).toString() );
        System.out.println( todos.get(0).getPropiedades().get(0).toString() );
        
        
        todos.set(0, new JugadorEspeculador( todos.get(0), 50) ) ;
        System.out.println( todos.get(0).toString() );
        System.out.println( todos.get(0).getPropiedades().get(0).toString() );
        
        System.out.println( CSO.ANSI_RED_BACKGROUND + CSO.ANSI_CYAN + "Son iguales los propietarios " + todos.get(0).getPropiedades().get(0).esEsteElPropietario(todos.get(0)) );
    
    }
    
}
